scipy/stats/_covariance.py

```diff
@@ -437,10 +477,13 @@ def _whiten(self, x):
 
     @cached_property
     def _covariance(self):
-        n = self._dimensionality
+        n = self._shape[-1]
         return (linalg.cho_solve((self._chol_P, True), np.eye(n))
                 if self._cov_matrix is None else self._cov_matrix)
 
+    def _colorize(self, x):
+        return linalg.solve_triangular(self._chol_P.T, x.T, lower=False).T
```

Comment:
Would this work for stacked inputs of shape `(x1, x2, ..., n, n)`? Or do you want to add support for that later?

Reply:
Good catch. We need to add it here, and it's easy to do. The tests were supposed to test that... I'll have to take a closer look.

Reply:
I confused myself. It does work and the test is working. I edited the code at the top to demonstrate. Please do try it!